src/commons/editor/Comments.tsx

```diff
@@ -0,0 +1,362 @@
+
+  // Determines if a messsage is being sent to server. For spinner.
+  const [loadingStatus, setLoadingStatus] = React.useState({} as { [id: string]: boolean });
+  const loadingStatusRef = React.useRef(loadingStatus);
+  loadingStatusRef.current = loadingStatus;
+
+  const {
+    updateComment,
+    updateCommentsBeingEdited,
+    removeComment,
+    removeCommentEdit,
+    isUnsubmittedComment
+  } = APIRef.current;
+
+  // Yes, this is an antipattern.
+  // It's also the only way to prevent the parent from being updated.
+  // While allowing the child to update.
+  const [commentsBeingEdited, setCommentsBeingEdited] = React.useState(
+    commentsBeingEditedRef.current
+  );
+
+  React.useEffect(() => {
+    const callback = (x: any) => {
+      setCommentsBeingEdited(x);
+    };
+    commentEditChangeEE.on('change', callback);
+    return () => {
+      commentEditChangeEE.off('change', callback);
+    };
+  }, [commentEditChangeEE, commentsBeingEditedRef]);
+
+  // ---------------- STATE HELPERS ----------------
+
+  // Will be required later to propagate the changes back.
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+
+  function updatePreviewCommentText(comment: IComment) {
+    return function (event: React.ChangeEvent<HTMLTextAreaElement>) {
+      const text = event.target.value;
+      updateCommentsBeingEdited({
+        ...comment,
+        text
+      });
+    };
+  }
+
+  // ---------------- CONTROLS ----------------
+
+  function cancelWithPrompt(comment: IComment) {
+    return function (evt: any) {
+      // eslint-disable-next-line no-restricted-globals
+      const sure = comment.text.trim() === '' || confirm('Are you sure about that?');
+      if (sure) {
+        if (isUnsubmittedComment(comment)) {
+          removeComment(comment);
+        } else {
+          removeCommentEdit(comment);
+        }
+      }
+    };
+  }
+
+  function confirmSubmit(comment: IComment) {
+    return async function (evt: any) {
+      // TODO: figure out a method for edits
+      const newComment: IComment = {
+        ...comment,
+        datetime: Date.now()
+      };
+
+      // This is to give a spinner to the users.
+      setLoadingStatus({
+        ...loadingStatus,
+        [comment.id]: true
+      });
+
+      try {
+        await sendToServer(newComment); // TODO: STUB FUNCTION, PLEASE UPDATE.
+        updateComment(newComment);
+        removeCommentEdit(newComment);
+        setErrorMsgs(omit(errorMsgs, [comment.id]));
+      } catch (e) {
+        setErrorMsgs({
+          ...errorMsgsRef.current,
+          [comment.id]: e
+        });
+      } finally {
+        // Pop quiz: why do we need to ref.current?
+        setLoadingStatus(omit(loadingStatusRef.current, [comment.id]));
+      }
+    };
+  }
+
+  function editComment(comment: IComment) {
+    return function (evt: any) {
+      // Puts this comment as being updated
+      updateCommentsBeingEdited(comment);
+    };
+  }
```

Comment:
Seems a bit verbose given its basically currying. Is this because of `tsx-no-lambda`?